go-controller/pkg/ovn/master.go

```diff
@@ -842,31 +848,122 @@ func (oc *Controller) addNodeAnnotations(node *kapi.Node, hostSubnets []*net.IPN
 	return nil
 }
 
-func (oc *Controller) addNode(node *kapi.Node) ([]*net.IPNet, error) {
-	oc.clearInitialNodeNetworkUnavailableCondition(node, nil)
-
-	hostSubnets, _ := util.ParseNodeHostSubnetAnnotation(node)
-	if hostSubnets != nil {
-		// Node already has subnet assigned; ensure its logical network is set up
-		return hostSubnets, oc.ensureNodeLogicalNetwork(node.Name, hostSubnets)
+func (oc *Controller) allocateNodeSubnets(node *kapi.Node) ([]*net.IPNet, []*net.IPNet, error) {
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node)
+	if err != nil {
+		// Log the error and try to allocate new subnets
+		klog.Infof("Failed to get node %s host subnets annotations: %v", node.Name, err)
```

Comment:
Actually just return an error here. This will cause a panic on L908 when doing `hostSubnets[:n]`, since  `ParseNodeHostSubnetAnnotation` always returns nil in the event of an error.

Reply:
Ah no, I am wrong. It won't panic.

Reply:
this is not really an error, it means we were not able to parse the annotation.
Previously it was totally ignored :shrug:

Reply:
Yes, I see your point. You're right

Reply:
I mean, you can justify one or the other, don't think anybody is wrong or right here ;)